### ISSUE TITLE
Replace DeviceComponentManager with NaturalOrderingManager to improve display order

### ISSUE THREAD

**Issue by tardoe** [digitalocean/netbox#2986]
### Environment
* Python version:  3.5.2
* NetBox version:  2.5.7

### Proposed Functionality
Replace the use of DeviceComponentManager in netbox/dcim/models.py with NaturalOrderingManager from the parent utilities/managers.py to better order display of device components.

### Use Case
This would affect the way device bays, front/rear ports, power connections and console connections are displayed to the user. Given all of these will most likely have a numerical component to the name, we have an opportunity to reuse the work done in the NaturalOrderingManager to make the above model's ordering consistent with Site, Device etc. views.

### Database Changes
None.

### External Dependencies
None.

**Comment by DanSheps** [comment]
DeviceComponentManager adjusts the sorting and enhances it to make the natural sorting better.  Could you explain how the current DeviceComponentManager is insufficient for the need for ordering?

**Comment by tardoe** [comment]
Thanks @DanSheps - I was going to extend on this but never got around to it.

For devices with Front/Rear ports - they tend to have the same naming characteristics as other device interfaces - that is, consecutive or patterned numerical names.

I patched my own installation such that RearPort and RearPortTemplate used the NaturalOrderingManager leaving FrontPort and FrontPortTemplate using the existing DeviceComponentManager - this was the result:

![Screenshot 2019-03-21 10 21 24](https://user-images.githubusercontent.com/5331531/54725648-fbfeda80-4bc3-11e9-88dc-40a5f20f8757.png)

![Screenshot 2019-03-21 10 21 18](https://user-images.githubusercontent.com/5331531/54725653-03be7f00-4bc4-11e9-8fe1-76e41a4eeb63.png)

I also tested this functionality with another familiar DC patch panel naming scheme used by $globalDCProvider with similar results.
`<city><number>:<level>:<function>:<cage>:<vertical>:<pair>`
